sqld/src/connection/mod.rs

```diff
@@ -266,14 +274,28 @@ impl<F: MakeConnection> MakeConnection for MakeThrottledConnection<F> {
         }
 
         let inner = self.connection_maker.create().await?;
-        Ok(TrackedConnection { permit, inner })
+        Ok(TrackedConnection {
+            permit,
+            inner,
+            atime: AtomicU64::new(now_millis()),
+        })
     }
 }
 
+#[derive(Debug)]
 pub struct TrackedConnection<DB> {
     inner: DB,
     #[allow(dead_code)] // just hold on to it
     permit: tokio::sync::OwnedSemaphorePermit,
+    atime: AtomicU64,
+}
+
+impl<DB> TrackedConnection<DB> {
+    pub fn idle_time(&self) -> Duration {
+        let now = now_millis();
+        let atime = self.atime.load(Ordering::Relaxed);
+        Duration::from_millis(now - atime)
+    }
```

Comment:
I'm not sure this is safe. SystemTime is not monotonic, so there is a chance that now is in the past compared to time, and `now - atime` may overflow

Reply:
Overall, since this is a temp fix, I would prefer that we work with `Mutex<tokio::time::Instant>`. The mutex should be completely uncontended, so it should be very cheap with parking lot mutex

Reply:
If our only concern here is overflow, I'd rather just do `saturating_sub`. I chose system time because it's the easiest to operate on timestamps, and since there's no AtomicInstant, I preferred that over a mutex, since it makes the code way simpler, and I'd rather keep temporary fixes dead simple.